fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/QueryPlanner.java

```diff
@@ -67,6 +67,18 @@
     @Nonnull
     RecordQueryPlan plan(@Nonnull RecordQuery query);
 
+    /**
+     * Create a plan to get the results of the provided query.
+     * This method returns a {@link QueryPlanResult} that contains the same plan ass returned by {@link #plan(RecordQuery)}
+     * with additional information provided in the {@link QueryPlanInfo}
+     *
+     * @param query a query for records on this planner's metadata
+     * @return a {@link QueryPlanResult} that contains the plan for the query with additional information
+     * @throws com.apple.foundationdb.record.RecordCoreException if the planner cannot plan the query
+     */
+    @Nonnull
+    QueryPlanResult planForQuery(@Nonnull RecordQuery query);
```

Comment:
I think the existing `plan` method is a verb. This looks like a noun (if it were a verb, I guess it'd mean prepare something in anticipation of this query). So I'd go with `planQuery` or `getPlanForQuery` or `buildPlanForQuery` of something like that.

Reply:
Done.